### PR TITLE
Include generated source files in the sourcepath of cyclefinder

### PR DESCRIPTION
**Pull request by confile**
Include generated source files in the sourcepath of cyclefinder

### CODE REVIEW

File: j2objc.gradle [hunk 1]
```diff
@@ -285,6 +285,15 @@ class J2objcCycleFinderTask extends DefaultTask {
             throw new InvalidUserDataException(message)
         }
 
+        def sourcepath = "${project.file("src/main/java").path}:${project.file("src/test/java").path}"
+	def generatedSourceDir = project.j2objcConfig.generatedSourceDir
```
Comment (brunobowden) [inline j2objc.gradle]: Indentation is different on the following lines compared to the previous one

File: j2objc.gradle [hunk 2]
```diff
@@ -285,6 +285,15 @@ class J2objcCycleFinderTask extends DefaultTask {
...
+	if (generatedSourceDir) {
+		println "include generatedSourceDir: ${generatedSourceDir}"
```
Comment (brunobowden) [inline j2objc.gradle]: I use 4 space indentation for blocks, this block uses 8

File: j2objc.gradle [hunk 3]
```diff
@@ -285,6 +285,15 @@ class J2objcCycleFinderTask extends DefaultTask {
...
+		def buildSrcFiles = project.files(project.fileTree(dir: generatedSourceDir, includes: ["**/*.java"]))
+		srcFiles += buildSrcFiles
+		sourcepath += ":${project.file(generatedSourceDir).path}"
+	}
```
Comment (brunobowden) [inline j2objc.gradle]: This code is repeated on lines 363-367. So it should be extracted to a common method. For example:

```
// Generated Files additions
srcFiles = J2objcUtils.addGeneratedFiles(srcFiles, project.j2objcConfig.generatedSourceDir)
sourcepath += ":${project.file(generatedSourceDir).path}"

srcFiles = J2objcUtils.fileFilter(srcFiles,
    ....
```

**Comment by brunobowden** [comment]
Thanks @confile, please make these changes and then I'll pull it in

**Comment by confile** [comment]
@brunobowden I did the changes.

File: j2objc.gradle [hunk 4]
```diff
@@ -234,6 +234,15 @@ class J2objcUtils {
             nameMap.put(file.name, file)
         }
     }
+    
+    static def addJavaFiles(Project proj, FileCollection files, String generatedSourceDir) {
```
Comment (brunobowden) [inline j2objc.gradle]: This is a generic function, so it should be sourceDir instead of generatedSourceDir as it could be used in other places.

**Comment by brunobowden** [comment]
Looking at it again, extracting the method didn't save that much but thanks for doing it. I'll merge now and then make the parameter name change myself.

File: j2objc.gradle (hunk 4 above)
Comment (confile) [inline j2objc.gradle]: @brunobowden done!

**Comment by confile** [comment]
your welcome!

**Pull request merged**